api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java

```diff
@@ -136,10 +138,14 @@ private int compareCostFractions(final double a, final double b) {
     return DoubleMath.fuzzyCompare(a, b, COST_FRACTION_TOLERANCE);
   }
 
-  private boolean expiredByCost(final DbUser user, final double currentCost) {
+  private boolean costAboveLimit(final DbUser user, final double currentCost) {
     return compareCosts(currentCost, getUserFreeTierDollarLimit(user)) > 0;
   }
 
+  private boolean costsDiffer(final double a, final double b) {
```

Comment:
It might help to have a small `MyCurrency` class to wrap the double, from which you could move to a standard representation using integral cents. Just to keep money from getting subtracted from weights or something.